tests/test_packages/test_connections/test_p2p_libp2p/test_aea_cli.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This test module contains tests for P2PLibp2p connection."""
+
+import functools
+import os
+import random
+import shutil
+import string
+import tempfile
+import time
+
+import pytest
+
+from aea.mail.base import Envelope
+from aea.multiplexer import Multiplexer
+from aea.protocols.default.message import DefaultMessage
+from aea.test_tools.test_cases import BaseAEATestCase
+
+from ....conftest import (
+    _make_libp2p_connection,
+    libp2p_log_on_failure,
+    skip_test_windows,
+)
+
+DEFAULT_PORT = 10234
+DEFAULT_DELEGATE_PORT = 11234
+DEFAULT_NET_SIZE = 4
+
+DEFAULT_LAUNCH_TIMEOUT = 25
+
+def with_new_agent(fn):
+    @functools.wraps(fn)
+    def wrapper(self, *args, **kwargs):
+        self._create_agent()
+        fn(self, *args, **kwargs)
+        self.delete_agents(self.agent_name)
+    return wrapper
```

Comment:
Plese use the functionality on the test cases and don't implement new functionality in the tests; the test should look and feel the same as all the other TestCases and any code not present in the testcase but needed should be added on the base class rather than here.